src/ScriptBuilder/ScriptGenerator.cs

```diff
@@ -129,7 +129,7 @@ void PurgeDialectDirs(string scriptPath)
 
     IEnumerable<string> GetSelectedDialects()
     {
-        return dialectOptions != null ? dialectOptions.Select(d => d.ToString()) : Enum.GetNames(typeof(BuildSqlDialect));
+        return dialectOptions.Any() ? dialectOptions.Select(d => d.ToString()) : Enum.GetNames(typeof(BuildSqlDialect));
```

Comment:
`dialectOptions` won't ever be null because it is initialized to an empty list in the ctor if it is passed in, so :+1:

Reply:
Given `dialectOptions` is a readonly collection I would prefer checking the count like it is done here https://github.com/Particular/NServiceBus.Persistence.Sql/blob/66f5ddec00ec000a444ac6e9f9287958df6ea7f1/src/ScriptBuilder/ScriptGenerator.cs#L98

Reply:
Updated. @danielmarbach do you want to quickly look again

Reply:
For those of us in the back, why is `.Count` better than `.Any()`?